mirgecom/initializers.py

```diff
@@ -40,6 +43,97 @@
 )
 from meshmode.dof_array import thaw
 from mirgecom.eos import IdealSingleGas
+from mirgecom.euler import split_conserved, join_conserved
+
+
+def _make_uniform_flow(x_vec, mass=1.0, energy=2.5, pressure=1.0,
+                       velocity=None, eos=IdealSingleGas()):
+    r"""Construct uniform, constant flow.
+
+    Construct a uniform, constant flow from mass, energy, pressure, and
+    an EOS object.
+
+    Parameters
+    ----------
+    x_vec: np.ndarray
+        Nodal positions
+    mass: float
+        Value to set `:math:\rho`
+    energy: float
+        Optional value to set `:math:\rho{E}`
+    pressure: float
+        Value to use for calculating `:math:\rho{E}`
+    velocity: np.ndarray
+        Optional constant velocity to set `:math:\rho\vec{V}`
+
+    Returns
+    -------
+    q: Object array of DOFArrays
+        Agglomerated object array with the uniform flow
+    """
+    dim = len(x_vec)
+    if velocity is None:
+        velocity = np.zeros(shape=(dim,))
+
+    _rho = mass
+    _p = pressure
+    _velocity = velocity
+    _gamma = eos.gamma()
+
+    mom0 = _rho * _velocity
+    e0 = _p / (_gamma - 1.0)
+    ke = 0.5 * np.dot(_velocity, _velocity) / _rho
+    x_rel = x_vec[0]
+    zeros = 0.0*x_rel
+    ones = zeros + 1.0
+
+    mass = zeros + _rho
+    mom = make_obj_array([mom0 * ones for i in range(dim)])
+    energy = e0 + ke + zeros
+
+    return flat_obj_array(mass, energy, mom)
+
+
+def _make_pulse(amp, r0, w, r):
+    r"""Create a Gaussian pulse.
+
+    The Gaussian pulse is defined by:
+
+    .. math::
+
+        G(\vec{r}) = a_0*\exp^{-(\frac{(\vec{r}-\vec{r_0})}{\sqrt{2}w})^{2}}\\
+
+    Where :math:`\vec{r}` is the nodal position, and the parameters are
```

Comment:
```suggestion
    Where :math:`\vec{r}` is the position, and the parameters are
```
The math doesn't know anything about nodes.